### PR TITLE
Fix custom banlists

### PR DESCRIPTION
**Pull request by HoeenCoder**
Travis will fail, the validator tests are failing but I don't have time left today to fix them. I'll work on them in the morning, in the mean time I opened this for review.

### CODE REVIEW

File: sim/dex.js [hunk 1]
```diff
@@ -746,7 +744,7 @@ class ModdedDex {
 			}
 			const subRuleTable = this.getRuleTable(subformat, depth + 1);
 			subRuleTable.forEach((v, k) => {
-				ruleTable.set(k, v || subformat.name);
+				if (!ruleTable.has('!' + k)) ruleTable.set(k, v || subformat.name);
```
Comment (Zarel) [inline sim/dex.js]: This is good, I'd been meaning to do this.

**Comment by HoeenCoder** [comment]
Ready to merge.

**Comment by sirDonovan** [comment]
This is good so far. Adding or removing rules still requires the prefix <code>Rule:</code> since <code>Dex.getSupplementaryBanlist()</code> was removed, which also broke formatting in displaying a custom banlist. We should re-implement that.

File: tournaments/index.js [hunk 2]
```diff
@@ -1010,7 +1010,7 @@ let commands = {
 			if (Monitor.countPrepBattle(connection.ip, connection)) {
 				return;
 			}
-			TeamValidator(tournament.teambuilderFormat).prepTeam(user.team).then(result => {
+			TeamValidator(tournament.teambuilderFormat, tournament.banlist).prepTeam(user.team).then(result => {
```
Comment (sirDonovan) [inline tournaments/index.js]: We also need to pass the banlist to the battles again for ruleset changes, see https://github.com/Zarel/Pokemon-Showdown/commit/fa1b45f7ab19cd420c28bc061e5ae14d4fd40df1.

**Comment by panpawn** [comment]
This is going to have to be rebased @HoeenCoder

**Comment by HoeenCoder** [comment]
Sorry, RL got busy and I didnt get to @sirDonovan s concerns, ill PR those when i get a chance to add them.

**Pull request merged**